rcgen/CHANGELOG.md

```diff
@@ -1,6 +1,101 @@
 
 # Changes
 
+## Release 0.13.0 - March XX, 2024
+
+Breaking changes:
+
+- The issuance API has been restructured to better match user expectation and to
+  avoid regenerating data when encoding to both DER and PEM.
+
+  Previously one created a `Certificate` with `Certificate::from_params`, and
+  serialized the `Certificate` with `Certificate::serialize_der`,
+  `Certificate::serialize_der_with_signer`, `Certificate::serialize_pem` and
+  `Certificate::serialize_pem_with_signer`. These functions have been removed.
+
+  With the updated API you create a `Certificate` directly from
+  `CertificateParams` using `CertificateParams::self_signed` or
+  `CertificateParams::signed_by`. You can serialize the `Certificate` to PEM or
+  DER using `Certificate::der()` or `Certificate::pem`.
+
+- `KeyPair` generation has been separted from certificate issuance. The
+  `Certificate::get_key_pair`, `Certificate::serialize_private_key_der` and
+  `Certificate::serialize_private_key_pem` fns have been removed. The
+  `Certificate` and `CertificateParams` types now only holds public data.
+  Issuance APIs now expect to be given both a `Certificate` and a signing
+  `KeyPair`, and the caller must generate their own `KeyPair` up-front using
+  `KeyPair::generate`, `KeyPair::generate_for`, or `KeyPair::generate_rsa_for`.
+
+- Certificate signing request creation and issuance has been reworked for
+  consistency with the updated API.
+
+  Previously a CSR could be created from a `Certificate` using
+  `Certificate::serialize_request_der` or `Certificate::serialize_request_pem`.
+  This is now done using `CertificateParams::serialize_request`. PEM/DER
+  serialization of the `CertificateSigningRequest` can be done with
+  `CertificateSigningRequest::der` or `CertificateSigningRequest::pem`.
+
+  Previously issuance of a certificate from a CSR could be done using
+  `CertificateSigningRequest::serialize_der_with_signer` or
+  `CertificateSigningRequest::serialize_pem_with_signer`. This is now done using
+  `CertificateSigningRequestParams::signed_by`, and serialization to DER/PEM can
+  be done by calling `Certificate::pem` or `Certificate::der` on the resulting
+  `Certificate`.
+
+- Throughout the API DER inputs are now represented using types from the Rustls
+  `pki-types` crate, e.g. `PrivateKeyDer`, `CertificateDer`,
```

Comment:
I think we should call it by its crates.io name, rustls-pki-types.

Reply:
Fixed.